namesys/publisher.go

```diff
@@ -159,6 +169,19 @@ func PutRecordToRouting(ctx context.Context, k ci.PrivKey, value path.Path, seqn
 	return nil
 }
 
+func waitOnErrChan(ctx context.Context, errs chan error) error {
+	select {
+	case err := <-errs:
+		if err != nil {
+			return err
+		}
```

Comment:
This `if` can be simplified to just `return err`, right? Then you can drop the final `return nil` too.